### PR TITLE
Refactor to a more understandable converter syntax

### PR DESCRIPTION
**Pull request by whikloj**
**JIRA Ticket**: none

# What does this Pull Request do?

Stops using the Converter interface and creates more developer friendly method names to make it clear what you are doing with the converter.

# Interested parties
@fcrepo4/committers

### CODE REVIEW

File: fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/api/rdf/HttpIdentifierConverter.java [hunk 1]
```diff
@@ -99,8 +98,13 @@ protected String doForward(final String httpUri) {
...
-    @Override
-    protected String doBackward(final String fedoraId) {
+    /**
+     * Convert an internal identifier to an external URI.
+     *
+     * @param fedoraId the internal identifier.
+     * @return the external URI.
+     */
+    protected String toExternalId(final String fedoraId) {
         LOGGER.debug(String.format("Translating Fedora ID %s to Http URI", fedoraId));
```
Comment (bbpennel) [inline fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/api/rdf/HttpIdentifierConverter.java]: I know its not part of this PR, but it'd still be best to switch to using `{}` notation for this log statement and the other ones in this file (its a minor performance issue)

File: fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/api/rdf/HttpIdentifierConverter.java [hunk 2]
```diff
@@ -80,12 +78,13 @@ public HttpIdentifierConverter(final UriBuilder uriBuilder) {
...
-
-    @Override
-    protected String doForward(final String httpUri) {
+    /**
+     * Convert an external URI to an internal ID.
+     *
+     * @param httpUri the external URI.
+     * @return the internal identifier.
+     */
+    protected String toInternalId(final String httpUri) {
```
Comment (bbpennel) [inline fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/api/rdf/HttpIdentifierConverter.java]: I think this has to be public, otherwise you won't be able to call it from outside the `http/api/rdf` package. Same with `toExternalId`

**Comment by awoods** [comment]
Thanks for this, @whikloj !
